### PR TITLE
implement --nobsdflags and --exclude-nodump

### PR DESCRIPTION
**Pull request by ThomasWaldmann**
see #3160 

could be also `--no-bsdflags`, but we already have `--noatime` (similar for names in the code).

### CODE REVIEW

**Comment by ThomasWaldmann** [comment]
any feedback?

**Comment by enkore** [comment]
> In general it is a good idea to look at exactly what is slow and what is not. For Borg, almost always Borg is slow, and not the syscalls. Only if genuinely the syscalls cause a problem an option should be added; otherwise, the code should be fixed.

That being said, it may be reasonable to remove NODUMP-based exclusion of files entirely for 1.1, even if that call would be a bit late.

**Comment by ThomasWaldmann** [comment]
OK, I'll add --exclude-nodump and it'll be off by default.

**Comment by ThomasWaldmann** [comment]
OK, so now there is --exclude-nodump (default: false, to avoid confusing/unexpected default behaviour) for people who want NODUMP-flagged files not being backed up.

--nobsdflags is for these cases:
- borg create: avoid backing up crappy bsdflags from source filesystems not supporting them or accidentally setting strange bsdflags (like that case with a macOS share having NODUMP everywhere).
- borg extract: if you have already backed up crappy bsdflags, you for sure do not want to have them restored.

File: src/borg/archiver.py [hunk 1]
```diff
@@ -2738,6 +2743,10 @@ def define_archive_filters_group(subparser, *, sort_by=True, first_last=True):
                               help='do not store atime into archive')
         fs_group.add_argument('--noctime', dest='noctime', action='store_true',
                               help='do not store ctime into archive')
+        fs_group.add_argument('--nobsdflags', dest='nobsdflags', action='store_true',
+                              help='do not read and store bsdflags (e.g. NODUMP, IMMUTABLE) into archive')
+        fs_group.add_argument('--exclude-nodump', dest='exclude_nodump', action='store_true',
+                              help='honour the NODUMP flag: do not store flagged files into archive')
```
Comment (enkore) [inline src/borg/archiver.py]: exclude files flagged NODUMP

Comment (enkore) [reply src/borg/archiver.py]: I think this option should go into the exclusion group.

**Comment by ThomasWaldmann** [comment]
@enkore if there's nothing else, I'ld collapse all changesets and merge.

**Pull request merged**